themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -1056,3 +1579,256 @@ public static Output<List<String>> split(Output<String> str) {
 {{% /choosable %}}
 
 {{< /chooser >}}
+
+## Common Pitfalls
+
+### String Interpolation
+
+When working with outputs and apply, you may see an error message like so:
+
+```
+Calling __str__ on an Output[T] is not supported. To get the value of an Output[T] as an Output[str] consider: 1. o.apply(lambda v: f"prefix{v}suffix") See https://pulumi.io/help/outputs for more details.
+```
+
+The reason this is happening is because the _Output_ value is being used before it has been resolved by from the API.
```

Comment:
I don't understand this suggestion, is the original sentence not semantically correct?

Reply:
Technically we return this error even if the output _has_ resolved, better to be consistent in the error than to have it working sometimes based on async races. Having said that I think I'd prefer to keep some form of your original wording rather than bringing "apply" into this.

Reply:
It's incorrect to say the user can call __str__ after it has been resolved. There's no way for them to do that in plain user code, they can use an "apply" to map over it, but they can't `await` the Output.

Reply:
To be really clear, I am nitpicking this:

> is being used before it has been resolved

So I can wait until after it's been resolved to call `__str__`? No!

Reply:
gonna leave this as is for now, feel free to follow up with another pr

Reply:
Happy to take up in a follow up PR.